include/User.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef USER_HPP
+# define USER_HPP
+
+#include <string>
+
+class User
+{
+	public:
+		User(const std::string &name);
+		User(const User &obj);
+		User &operator=(const User &rhs);
+		~User(void);
+
+		const std::string	&getName(void) const;
+	
+	private:
+		std::string	_name;
```

Comment:
The `User` class should have a reference to a Peer and the name would be stored in it. It prevents duplication of code.